### PR TITLE
refactor relatedpackages logic

### PR DESCRIPTION
**Pull request by joaoopereira**
- chore: reduce log messages verbosity set some log messages to debug
- chore(refactor): improve RelatedPackages logic

### CODE REVIEW

File: cmf-cli/Commands/new/IoTCommand.cs [hunk 1]
```diff
@@ -187,7 +188,7 @@ public void ExecuteV10(IDirectoryInfo workingDir, string version, string htmlPac
 
             iotCustomPackage.RelatedPackages = new()
             {
-                new RelatedPackage() { Path = fileSystem.DirectoryInfo.New(this.fileSystem.Path.GetRelativePath(iotCustomPackageWorkDir.FullName, fileSystem.DirectoryInfo.New(htmlPackageLocation).FullName)), PreBuild = true, PrePack = true }
+                new RelatedPackage() { Path = Path.Join(fileSystem.Path.GetRelativePath(iotCustomPackageWorkDir.FullName, fileSystem.DirectoryInfo.New(htmlPackageLocation).FullName)), PreBuild = true, PrePack = true }
```
Comment (m-s-) [inline cmf-cli/Commands/new/IoTCommand.cs:191]: should use IO.Abstractions

File: cmf-cli/Commands/new/IoTCommand.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 using System.Collections.Generic;
 using System.CommandLine;
 using System.CommandLine.NamingConventionBinder;
+using System.IO;
```
Comment (m-s-) [inline cmf-cli/Commands/new/IoTCommand.cs:11]: Absolutely not :P

File: cmf-cli/Handlers/PackageType/SecurityPortalPackageTypeHandler.cs [hunk 3]
```diff
@@ -79,7 +79,7 @@ public override void Pack(IDirectoryInfo packageOutputDir, IDirectoryInfo output
 
                 FinalArchive(packageOutputDir, outputDir);
 
-                Log.Information($"{outputDir.FullName}/{CmfPackage.ZipPackageName} created");
+                Log.Debug($"{outputDir.FullName}/{CmfPackage.ZipPackageName} created");
```
Comment (m-s-) [inline cmf-cli/Handlers/PackageType/SecurityPortalPackageTypeHandler.cs:82]: missing the Log.Information here for consistency?

File: cmf-cli/Commands/new/IoTCommand.cs (hunk 1 above)
Comment (m-s-) [reply cmf-cli/Commands/new/IoTCommand.cs:191]: also, are these equivalent?

File: core/Objects/CmfPackage.cs [hunk 4]
```diff
@@ -647,9 +648,44 @@ public static CmfPackage Load(IFileInfo file, bool setDefaultValues = false, IFi
...
+            string fileContent = FileInfo.ReadToString();
+            JsonConvert.PopulateObject(fileContent, this);
+            IsToSetDefaultValues = setDefaultValues;
+            Location = PackageLocation.Local;
+
+            RelatedPackages?.Load(this);
+
+        }
+
+        public void Pick()
```
Comment (m-s-) [inline core/Objects/CmfPackage.cs:672]: should be Peek unless I'm misunderstanding the purpose of the method

Comment (joaoopereira) [reply core/Objects/CmfPackage.cs:672]: you are correct

File: cmf-cli/Handlers/PackageType/SecurityPortalPackageTypeHandler.cs (hunk 3 above)
Comment (joaoopereira) [reply cmf-cli/Handlers/PackageType/SecurityPortalPackageTypeHandler.cs:82]: yap

File: cmf-cli/Commands/new/IoTCommand.cs (hunk 1 above)
Comment (joaoopereira) [reply cmf-cli/Commands/new/IoTCommand.cs:191]: maybe not, but I found a simple way to get this

**Review by m-s- — APPROVED** [review]

**Pull request merged**